### ISSUE TITLE
Getting warning about HDF5 dependency

### ISSUE THREAD

**Issue by emstoudenmire** [ITensor/ITensors.jl#173]
I am getting a warning "Warning: Package ITensors does not have HDF5 in its dependencies" now, even though Project.toml has HDF5 listed under [deps] with the correct uuid. I also took the step of doing "activate ITensors.jl" in the pkg system, and adding HDF5 to update, which regenerated Manifest.toml. But I still get the same warning afterward.

**Comment by orialb** [comment]
This doesn't happen for me in a fresh project environment where only ITensors has been added  (i.e `using ITensors` does not output any warning). 
I do see it in my default Julia 1.1 environment, running `Pkg.resolve()` is supposed to solve it, but my specific default 1.1 env is too messy at the moment for resolve to work (due to different package version restrictions colliding). 
In general with new projects I try to work only with project specific environments to prevent the possibility of this kind dependency mess. 

TL;DR: Did you try running `Pkg.resolve()` in the environment from which you are loading ITensors.jl ?

**Comment by orialb** [comment]
Just to clarify that this always happens when you `dev` a package and add a dependency to it, as explained in [Pkg.jl docs](https://julialang.github.io/Pkg.jl/v1.3/managing-packages/#Developing-packages-1):

> Note that if you add a dependency to a package that tracks a local path, the Manifest (which contains the whole dependency graph) will become out of sync with the actual dependency graph. This means that the package will not be able to load that dependency since it is not recorded in the Manifest. To synchronize the Manifest, use the REPL command resolve.

**Comment by emstoudenmire** [comment]
Thanks Ori. This is very helpful. I did do `using Pkg; Pkg.resolve()` and that made the warning go away.

**Comment by orialb** [comment]
By the way you can also just run `resolve` in the REPL pkg mode.

**Comment by emstoudenmire** [comment]
I see, thank. Yes, Julia needs to update some of its warning messages to reflect the new pkg system.
